### PR TITLE
Update README and Documentation for GeoTrellis 3.0

### LINKED ISSUE #3016: Docs for module structure need updating
https://github.com/locationtech/geotrellis/blob/master/docs/guide/module-hierarchy.rst needs to be brought up to snuff following addition/reconfiguration to `geotrellis.layer` and `geotrellis.store`

### PR DESCRIPTION
**Pull request by CloudNiner**
## Overview

**NOTE:** This shouldn't be merged until we're ready to cut the 3.0 release, otherwise we'll have incorrect information for GT 2.3 in the project README.

This is not intended to be a comprehensive update/rewrite. The intention of this PR is to streamline the information presented in the README and better highlight additional sources of information. It also intends to close a few specific outstanding issues, starting with #3016.

### Checklist

- [x] `docs/CHANGELOG.rst` updated, if necessary
- [x] `docs` guides update, if necessary
- [-] New user API has useful Scaladoc strings
- [-] Unit tests added for bug-fix or new feature

### Notes

Other fixes:
- Added missing pipeline guide to the sidebar
- Updated version and copyright in docs config
- Fixed minor issue where we don't specify python 3 in the docs `make serve` command

Closes #3016

### CODE REVIEW

**Comment by CloudNiner** [comment]
@pomadchin this is ready for review. I think I've gotten as far on this as I will for the 3.0 release.

File: docs/tutorials/setup.rst [hunk 1]
```diff
@@ -84,10 +82,10 @@ the ``libraryDependencies`` list in your ``build.sbt``:
...
 
     libraryDependencies ++= Seq(
-        "org.locationtech.geotrellis" %% "geotrellis-spark"  % "2.3.1",
-        "org.locationtech.geotrellis" %% "geotrellis-s3"     % "2.3.1", // now we can use Amazon S3!
-        "org.apache.spark"            %% "spark-core"        % "2.4.1" % "provided",
-        "org.scalatest"               %% "scalatest"         % "3.0.0" % "test"
+        "org.locationtech.geotrellis" %% "geotrellis-spark"    % "3.0.0",
+        "org.locationtech.geotrellis" %% "geotrellis-s3-spark" % "3.0.0", // now we can use the Amazon S3 store!
+        "org.apache.spark"            %% "spark-core"          % "2.4.1" % "provided",
+        "org.scalatest"               %% "scalatest"           % "3.0.0" % "test"
```
Comment (pomadchin) [inline docs/tutorials/setup.rst]: Hm I noticed that these are outdated, I will update them.

Comment (CloudNiner) [reply docs/tutorials/setup.rst]: What is out of date here? I believe the geotrellis and spark dependencies are up to date as listed here.

Comment (pomadchin) [reply docs/tutorials/setup.rst]: @CloudNiner 

```scala
"org.apache.spark"            %% "spark-core"  % "2.4.3" % Provided,
"org.scalatest"                    %% "scalatest"     % "3.0.8" % Test
```

Comment (CloudNiner) [reply docs/tutorials/setup.rst]: Whoops. I stand corrected. Updated.

**Comment by pomadchin** [comment]
Merging this one after #3128 to cut down a release.

**Pull request merged**